test.sh

```diff
@@ -106,9 +106,16 @@ function save_docker_image {
     then
         zcat $imagefile | docker load
     fi
+    imageid=$(docker images -q --no-trunc "$imagename")
     build_docker_image "$dockerfile" "$imagename"
-    docker save "$imagename" | gzip > "$imagefile"
-    return "${PIPESTATUS[0]}"
+    newimageid=$(docker images -q --no-trunc "$imagename")
+    rc=0
+    if [ "$imageid" != "$newimageid" ]
+    then
+        docker save "$imagename" | gzip > "$imagefile"
+        rc="${PIPESTATUS[0]}"
+    fi
+    return "$rc"
```

Comment:
Exactly. Not a big deal.

On Tuesday, December 15, 2015, Nathaniel Cook <notifications@github.com>
wrote:

> In test.sh
> <https://github.com/influxdb/influxdb/pull/5126#discussion_r47726830>:
>
> >      build_docker_image "$dockerfile" "$imagename"
> > -    docker save "$imagename" | gzip > "$imagefile"
> > -    return "${PIPESTATUS[0]}"
> > +    newimageid=$(docker images -q --no-trunc "$imagename")
> > +    rc=0
> > +    if [ "$imageid" != "$newimageid" ]
> > +    then
> > +        docker save "$imagename" | gzip > "$imagefile"
> > +        rc="${PIPESTATUS[0]}"
> > +    fi
> > +    return "$rc"
>
> Its not necessary, but is in general a good practice. If somehow rc
> contained whitespace than it would pass two arguments to return and break.
> But since rc is either explicitly 0 or the result of the pipe call its not
> really possible for it to contain whitespace.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/influxdb/influxdb/pull/5126/files#r47726830>.
>